DatadogInternal/Sources/Utils/Sampler.swift

```diff
@@ -21,3 +30,25 @@ public struct Sampler {
         return Float.random(in: 0.0..<100.0) < samplingRate
     }
 }
+
+/// A sampler that determines sampling decisions deterministically (the same each time).
+public struct DeterministicSampler: Sampling {
```

Comment:
for inspiration, I think we can multiple samplers like here https://github.com/open-telemetry/opentelemetry-swift/blob/4eb75bc8f0d6449bc197637b5c6044b51ab8c4ed/Sources/OpenTelemetrySdk/Trace/Samplers/Samplers.swift#L10

I'm particularly interested in `ParentBasedSampler`

Reply:
@ganeshnj I'm sure what is the actual ask here 🙂, could you clarify? The SDK doesn't expose any sampler implementation to the end-user (interface + impls exist only in `DatadogInternal`). Because end-users only work with `sampleRate: Float` API today, I don't see reason for introducing more samplers than we need for SDK internals.

Reply:
apologies, it wasn't clear. I mean to say - for our internal working we can use some existing pattern. I wasn't expecting any action on the comment. Just for information if we have tricky cases around sampling decision.